documentation/adr/004-multi-LZA-configuration-management.md

```diff
@@ -0,0 +1,44 @@
+---
+layout: page.njk
+status: Proposed
+date: git Created
+tags:
+  - adr
+order: 4
+title: 004 - Enable Multi LZA Configuration 
+---
+
+**Date: {{ page.date.toISOString().substring(0,10) }}**
+
+**Status: {{ status }}**
+
+## Context
+
+To independently develop and assure features and functionality before general availability, multiple LZA environments are required as part of the teams SDLC. As a result, the team need a way to manage environment specific configuration e.g., workload accounts.
+
+The files used within LZA to manage the configuration of the deployment are YAML format, with the exception being IAM policies that are JSON. These are not required to differ between deployments.
+
+Having discussed this requirement with AWS, their recommendation is to use templating as part of the deployment pipeline to pass in environment specific config.
+
+## Decision
+
+We will implement templating before deployment to CodeCommit.
+
+We will move to having a baseline configuration, with environment specific code to be applied on top.
+
+We will ensure differences in configuration can be validated and approved before merge, and deployment to an environment.
+
+We will ensure that baseline configuration and deployed configuration is visible within CodeCommit to platform engineers.
+
+We will create a new process for platform engineers to follow to ensure changes are deployed correctly between environments.
+
```

Comment:
Did we also say we would prevent access being given to the repo outside of LZA i.e. we don't want a platform to be able to give access to the repository outside of LZA. i.e. we need to centrally manage access to this repository

Reply:
Updated in latest commit.